src/Storages/StorageS3.cpp

```diff
@@ -198,14 +204,18 @@ class StorageS3Source::DisclosedGlobIterator::Impl : WithContext
             if (block.has("_file"))
                 file_column = block.getByName("_file").column->assumeMutable();
 
-            for (const auto & row : result_batch)
+            for (const auto & key_info : result_batch)
             {
-                const String & key = row.GetKey();
+                const String & key = key_info.GetKey();
                 if (recursive || re2::RE2::FullMatch(key, *matcher))
                 {
                     String path = fs::path(globbed_uri.bucket) / key;
+                    const size_t key_size = key_info.GetSize();
+                    total_size += key_size;
```

Comment:
`total_size` should not be updated for keys that will be filltered by virtual columns. That the reason why test has failed.